packages/core/src/components/forms/numericInput.tsx

```diff
@@ -477,22 +483,26 @@ export class NumericInput extends AbstractComponent<HTMLInputProps & INumericInp
     }
 
     private invokeOnValueChangeCallback(value: string) {
-        const valueAsString = value;
-        const valueAsNumber = +value; // coerces non-numeric strings to NaN
-        Utils.safeInvoke(this.props.onValueChange, valueAsNumber, valueAsString);
+        Utils.safeInvoke(this.props.onValueChange, +value, value);
+    }
+
+    private invokeOnButtonClickCallback(value: string) {
+        Utils.safeInvoke(this.props.onButtonClick, +value, value);
```

Comment:
ok let's get clever and do both in one method:

```ts
private invokeValueCallback(value: string, callback = this.props.onButtonClick) {
    Utils.safeInvoke(callback, +value, value);
}
```

types of oBC and oVC are the same so by assigning a default value you don't have to redeclare the function signature here!